src/geode/mesh/core/surface_mesh.cpp

```diff
@@ -319,7 +355,8 @@ namespace geode
         const auto vertex = polygon_vertex.vertex_id;
         const auto polygon = polygon_vertex.polygon_id;
         const auto nb_vertices = nb_polygon_vertices( polygon );
-        return { polygon, ( vertex + 1 ) % nb_vertices };
+        return { polygon,
+            static_cast< local_index_t >( ( vertex + 1 ) % nb_vertices ) };

```

Comment:
Cast are required when using arithmetic computation because unsigned char are promoted to int, so we need to narrow down the type.